src/mds/StrayManager.cc

```diff
@@ -637,23 +637,29 @@ void StrayManager::eval_remote_stray(CDentry *stray_dn, CDentry *remote_dn)
 {
   assert(stray_dn != NULL);
   assert(stray_dn->get_dir()->get_inode()->is_stray());
+  CDentry::linkage_t *stray_dnl = stray_dn->get_projected_linkage();
```

Comment:
Are stray_dnl/stray_in just moved to the outer scope in order to run the asserts in more cases?